### PR TITLE
Fix missing models

### PR DESCRIPTION
**Pull request by SamGondelman**
Fixes [FB7726](https://highfidelity.fogbugz.com/f/cases/7726/Model-entities-not-rendering-when-they-should)

Moves model loading logic from renderUpdate, which is only called when entities change, to a new RenderableEntityItem::update(), which is called as often as it is needed.  This fixes a serious bug with model loading that is hidden in master by the fact that the entity server sends entities repeatedly as your view changes.

Also, models are properly marked as just meta now, instead of meta and shape.

Also, if the model cache already contains a model file, it will still fire the finished signal.

Test plan:
- Download [the new entity server PR](https://github.com/highfidelity/hifi/pull/11141).  Run sandbox from *that PR* NOT this one.  The issue is extremely obvious with the new server logic, and rarely noticeable in master.
- Download a content set with a good number of model entities.  Make sure there are multiple models with exactly the same fbx file and at least some models that have parent model entities.  The [dev-welcome content](http://dev-welcome.highfidelity.io:8083/) is pretty good, but it would be good to test other content sets too.
- All model entities should load normally.  Reload your content (Edit -> Reload Content) while looking at one part of the domain and then turn to look behind you.  No models should be missing once all your downloads finish.
- If model A is a child of model B and you look at only model A (model B is out of view and at least several meters away) and reload content, model A should not show up.  Turn to look at model B, and then look back at model A.  Now model A should show up.
- If model A and model B have the same model file and you look at only model A (model B is out of view and at least several meters away), reload content, and wait a while so that everything has loaded, when you turn to look at model B it should appear.
- Edit a model's properties, like visible, textures, and model URL.  All changes should take effect as expected.
- Test on a server running master, like dev-welcome.  All content should appear.
- Performance smoke test.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3541/HighFidelity-Beta-PR11436-9f12d3a3658298c6afb7d6a27b65735fdd4e4548.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3541/HighFidelity-Beta-PR11436-9f12d3a3658298c6afb7d6a27b65735fdd4e4548.exe

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -222,6 +222,14 @@ void EntityTreeRenderer::updateChangedEntities(const render::ScenePointer& scene
...
     }
 
+    if (!_entitiesInScene.empty()) {
+        for (const auto& entry : _entitiesInScene) {
+            const auto& renderable = entry.second;
+            if (renderable) {
+                renderable->update(scene, transaction);
+            }
+        }
+    }
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: The `if (!_entitiesInScene.empty())` check seems unnecessary: it is almost always `true` and even when it is `false` the `for` loop is more or less a no-op.

Maybe we should skip this looping when `changedEntities` is empty instead?  That might save a few CPU cycles.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3555/HighFidelity-Beta-PR11436-be27376c52d3b37106383a8cab1ab458f3f4d6e1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3555/HighFidelity-Beta-PR11436-be27376c52d3b37106383a8cab1ab458f3f4d6e1.exe

**Comment by Nex-Pro** [comment]
T E S T I N G this

**Comment by Nex-Pro** [comment]
Awaiting fresh build

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3563/HighFidelity-Beta-PR11436-ea8cbe55b72ffc60187557bacd489f06cf4d7f07.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3563/HighFidelity-Beta-PR11436-ea8cbe55b72ffc60187557bacd489f06cf4d7f07.dmg

**Comment by Nex-Pro** [comment]
T E S T I N G this

**Comment by Nex-Pro** [comment]
Running this with sandbox from 11141

**Comment by Nex-Pro** [comment]
Tested on win10

Results: **SEE NOTES**


Download the new entity server PR. 

Run sandbox from that PR NOT this one. 

The issue is extremely obvious with the new server logic, and rarely noticeable in master.

Download a content set with a good number of model entities. **USED medeival village in localhost as it has a lot of models as per requirement**

Make sure there are multiple models with exactly the same fbx file and at least some models that have parent model entities. **Parented some trees to another in localhost parented some EZ Blocks at dev-welcome**
The dev-welcome content is pretty good, but it would be good to test other content sets too.

All model entities should load normally. **SUCCESS** 

Reload your content (Edit -> Reload Content) while looking at one part of the domain and then turn to look behind you. 
No models should be missing once all your downloads finish. **SUCCESS**

If model A is a child of model B and you look at only model A (model B is out of view and at least several meters away) and reload content, model A should not show up. **Success at dev-welcome FAIL in 11141 it shows up every time** 
Turn to look at model B, and then look back at model A. Now model A should show up. **SUCCESS at dev welcome Did not need to in 11141**

If model A and model B have the same model file and you look at only model A (model B is out of view and at least several meters away), reload content, and wait a while so that everything has loaded, when you turn to look at model B it should appear. **Success at dev-welcome FAIL in 11141** 

Edit a model's properties, like visible, textures, and model URL. All changes should take effect as expected. **SUCCESS**

Test on a server running master, like dev-welcome. All content should appear. **SUCCESS**

Performance smoke test. **Seems fine**

**Comment by SamGondelman** [comment]
Just to clarify: there is a separate bug in master, sometimes, where parent entity query AA boxes do not surround their children, which is why I mentioned that you might have to turn to see a parent before the child shows up.  The fact that piper didn't have that bug was different than what I expected, but is actually better behavior, and is unrelated to this PR anyway.  Models showed up, so this is a pass!

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 2]
```diff
@@ -1243,6 +1247,12 @@ void ModelEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& sce
     }
 }
 
+void ModelEntityRenderer::modelLoaded(bool success) {
+    if (success) {
+        _modelJustLoaded = true;
+    }
+}
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: I think maybe this should be called: `handleModelLoaded()` or `onModelLoaded()` (something that better suggests what it is supposed to do).

When I first saw it I thought it was supposed to be: `isModelLoaded()` but... it was changing something and returned `void`!

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3575/HighFidelity-Beta-PR11436-86198606e036f7f5cafbc75295983fd0c5421a5a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3575/HighFidelity-Beta-PR11436-86198606e036f7f5cafbc75295983fd0c5421a5a.dmg

**Pull request merged**